### ISSUE TITLE
Closure type cannot be inferred with parameters

### ISSUE THREAD

**Issue by Frotty** [wurstscript/WurstScript#923]
This works:

```
package ClosureTest

interface A
	function foo() returns string

interface B
	function foo() returns int

function bar(A a)

function bar(B b)

init
	bar(() -> "")
```

This does not:
```
interface A
	function foo(int lvl) returns string

interface B
	function foo(int lvl) returns int

function bar(A a)

function bar(B b)

init
	bar(lvl -> "")
```

=>
```
Call to function bar is ambiguous. Alternatives are:
  * function bar defined in line 9 (C:\Users\Frotty\Documents\GitHub\WurstStdlib2\wurst\objediting\ClosureTest.wurst)
 * function bar defined in line 11 (C:\Users\Frotty\Documents\GitHub\WurstStdlib2\wurst\objediting\ClosureTest.wurst)

Could not infer type for parameter lvl. The target type could not be uniquely determined for expected type (A or B).

Closures can only be used when a interface or class type is given, but at this position a (A or B) is expected.
```

**Comment by peq** [comment]
The current rule is like this:
If all closure parameter types are given explicitly, the closure is checked first and then overloading is resolved with the resulting function type.

If the parameter types are not given, they need to be inferred from the context and the context does not give a unique closure type here.  

In this particular case, we could maybe add a special case because the parameter type must always be int.

tldr; just write `bar((int lvl) -> "")`
